lib/hackerone/client/report.rb

```diff
@@ -75,6 +75,32 @@ def activities
         end
       end
 
+      def program
+        @program || Program.find(relationships[:program][:data][:attributes][:handle])
+      end
+
+      def assign_to_user(name)
+        fail(
```

Comment:
I think I understand what you mean. Generally I agree with you, it just so happens that the HackerOne API here isn't very helpful with the errors that are provided.

For example, if I try to assign a report as with a read-only API token (e.g. the token is not part of any groups), it will return this JSON:

```json
{"errors":[{"status":403}]}
```

it would be helpful if the API could provide some clue as to what is going on. For example, something like this:

```json
{"errors":[{"status":403, "reason": "You need to be part of a group with the report management permission to assign a report."}]}
```

The same goes for assigning to users that cannot be assigned to:

```json
{"errors":[{"status":500}]}
```

And for assigning to users that are not part of your team, or don't exist at all:

```json
{"errors":[{"status":404}]}
```

Again, I would love to see the API return some kind of helpful error message. As it does not, I've implemented that in this gem instead. I'll leave it to @oreoshake whether or not that's a reasonable compromise -- I think there's something to say for going either way.